python_ta/__init__.py

```diff
@@ -92,8 +95,26 @@ def _check(module_name='', level='all', local_config='', output=None):
                 linter.check(file_py)  # Lint !
                 current_reporter.print_messages(level)
                 current_reporter.reset_messages()  # Clear lists for any next file.
+                f_paths.append(file_py)  # Appending paths for (potential) upload
                 print('[INFO] File: {} was checked using the configuration file: {}'.format(
                     file_py, linter.config_file))
+
+            if linter.config.pyta_error_permission or linter.config.pyta_file_permission:
+                errs = [msg for msg in current_reporter.messages_by_file if msg.filename in f_paths]
+                if linter.config.pyta_file_permission:
+                    upload_to_server(errors=errs,
+                                     config=linter.config.__dict__,
+                                     default=reset_linter().config.__dict__,
+                                     url=linter.config.pyta_server_address,
+                                     version=__version__,
+                                     paths=f_paths)
+                else:
```

Comment:
I want the two options to be completely independent of each other. The best way to structure this code is:

```
# initialize default variables for errors and paths
if linter.config.pyta_file_permission:
  # set paths
if linter.config.pyta_error_permission:
  # set errors

# call upload_to_server
```